### ISSUE TITLE
`deno repl` Uncaught ReferenceError: __DENO_NODE_GLOBAL_THIS_XXX__ is not defined

### ISSUE THREAD

**Issue by loynoir** [denoland/deno#18450]
### Reproduce
```sh
$ tail -n +1 *
==> app.mts <==
await import('reproduce')

==> deno.jsonc <==
{
  "imports": {
    "ip-regex": "npm:ip-regex@5.0.0",
    "reproduce": "./reproduce.mts"
  }
}

==> reproduce.mts <==
import ipRegex from 'ip-regex'
console.log({ ipRegex })
```

### Expected
Application import is OK. And repl import is also OK.

```sh
$ deno run -A app.mts 
{
  ipRegex: [Function: ipRegex] { v4: [Function (anonymous)], v6: [Function (anonymous)] }
}
```

### Actual
Repl import got .`Uncaught ReferenceError: __DENO_NODE_GLOBAL_THIS_XXX__ is not defined`

```sh
$ deno repl -A
Deno 1.32.1
exit using ctrl+d, ctrl+c, or close()
> await import('reproduce')
Uncaught ReferenceError: __DENO_NODE_GLOBAL_THIS_XXX__ is not defined
    at file:///home/vscode/.cache/deno/npm/registry.npmjs.org/ip-regex/5.0.0/index.js:1:18
```

**Comment by GJZwiers** [comment]
Did a bisect and found c6c8c91a6 introduced this bug.

**Comment by bartlomieju** [comment]
Solution to this problem is to always initialize the Node compat layer when entering the REPL.

**Comment by adamgreg** [comment]
Is there any kind of workaround for this at the moment?

**Comment by bartlomieju** [comment]
Sadly no, this will be fixed by https://github.com/denoland/deno/pull/19307 in a few days.

**Comment by adamgreg** [comment]
Oh that's great! Thanks

**Comment by DMeechan** [comment]
I ran into the same "ReferenceError: \_\_DENO_NODE_GLOBAL_THIS_0_48_0__ is not defined" error when importing [nanoid](https://github.com/ai/nanoid) into my Deno Fresh project via npm:

```javascript
import { nanoid } from "npm:nanoid@4.0.2";

nanoid();

// Results in error:
error: Uncaught (in promise) ReferenceError: __DENO_NODE_GLOBAL_THIS_0_48_0__ is not defined
    at file:///my_project_name/node_modules/.deno/nanoid@4.0.2/node_modules/nanoid/url-alphabet/index.js:1:18
```

I'm not sure why, but importing nanoid from [esm.sh](https://esm.sh/) solved the error for me:

```javascript
import { nanoid } from "https://esm.sh/nanoid@4.0.2";

nanoid(); // success
```

Perhaps Deno used a different set of globals when importing from esm.sh?

**Comment by bartlomieju** [comment]
This is now fixed in Deno v1.36.0
